.github/workflows/ubuntu-ci.yml

```
@@ -0,0 +1,73 @@
+
+name: Fast DDS Docs Ubuntu CI
+
+on:
+  workflow_dispatch:
+    inputs:
+      os-version:
+        description: 'OS version to run the workflow'
+        required: false
+        default: 'ubuntu-22.04'
+        type: string
+      colcon-args:
+        description: 'Extra arguments for colcon cli'
+        required: false
+        type: string
+      cmake-args:
+        description: 'Extra arguments for cmake cli'
+        required: false
+        type: string
+      ctest-args:
+        description: 'Extra arguments for ctest cli'
+        required: false
+        type: string
+      fastdds-docs-branch:
+        description: 'Branch or tag of Fast DDS Docs repository'
+        type: string
+        required: true
+      fastdds-branch:
+        description: 'Branch or tag of Fast DDS repository'
+        type: string
+        required: true
+      fastdds-python-branch:
+        description: 'Branch or tag of Fast DDS Python repository'
+        required: true
+        type: string
+      run-tests:
+        description: 'Run test suite of Fast DDS python'
+        required: false
+        type: boolean
+        default: true
+      use-ccache:
+        description: 'Use CCache to speed up the build'
+        required: false
+        type: boolean
+        default: false
+
+  pull_request:
+    types:
+      - review_requested
+    paths-ignore:
+      - '**.md'
+      - '**.txt'
+      - '!**/CMakeLists.txt'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  ubuntu-ci:
+    uses: ./.github/workflows/reusable-ubuntu-ci.yml
+    with:
+      os-version: ${{ inputs.os-version || 'ubuntu-22.04' }}
+      label: 'ubuntu-ci'
+      colcon-args: ${{ inputs.colcon-args }}
+      cmake-args: ${{ inputs.cmake-args }}
+      ctest-args: ${{ inputs.ctest-args }}
+      fastdds-docs-branch: ${{ inputs.fastdds-docs-branch || github.ref || 'master' }}
```

Master is not needed because if this workflow is ran from a pr, `github.ref` branch exists, if it's ran as a workflow dispatch the `inputs.shapes-demo-branch` is mandatory
```suggestion
      fastdds-docs-branch: ${{ inputs.fastdds-docs-branch || github.ref }}
```